wallets/index.js

```diff
@@ -73,8 +73,8 @@ export function useWallet (name) {
       // validate should log custom INFO and OK message
       // validate is optional since validation might happen during save on server
       // TODO: add timeout
-      await wallet.validate?.(newConfig, { me, logger })
-      await saveConfig(newConfig)
+      const validConfig = await wallet.validate?.(newConfig, { me, logger })
+      await saveConfig(validConfig ?? newConfig)
```

Comment:
This is the most invasive change. It allows `validate` to return any modifications to the config before storing. `lnc` needs to store diffie helman keys, so when validate generates them, we return them in the config.